compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/model/BirScope.java

```diff
@@ -0,0 +1,35 @@
+/*
+ *  Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ *
+ */
+
+package org.wso2.ballerinalang.compiler.bir.model;
+
+/**
+ * Hold the scope of the instruction w.r.t variable declarations.
+ *
+ * @since 2.0.0
+ */
+public class BirScope {
+    public int id;
+    public BirScope parent;
```

Comment:
shall we have getter setter instead of having public variables?

Reply:
This is used as a data class without any functionality. Using getters setters would be too much imo. I'll make these final instead.